doc/how_to/custom_components/esm/panes.md

```diff
@@ -0,0 +1,311 @@
+    removeCy();
+
+    const div = document.createElement('div');
+    div.style.width = "100%";
+    div.style.height = "100%";
+    // Cytoscape raises warning of position is static
+    div.style.position = "relative";
+
+    model.on('after_render', () => {
+        cy = cytoscape({
+            container: div,
+            layout: {name: model.layout},
+            elements: model.object,
+            zoom: model.zoom,
+            pan: model.pan
+        })
+        cy.style().resetToDefault().append(model.style).update()
+        cy.on('select unselect', function (evt) {
+            model.selected_nodes = cy.elements('node:selected').map(el => el.id())
+            model.selected_edges = cy.elements('edge:selected').map(el => el.id())
+        });
+
+        model.on('object', () => {cy.json({elements: model.object});cy.resize().fit()})
+        model.on('layout', () => {cy.layout({name: model.layout}).run()})
+        model.on('zoom', () => {cy.zoom(model.zoom)})
+        model.on('pan', () => {cy.pan(model.pan)})
+        model.on('style', () => {cy.style().resetToDefault().append(model.style).update()})
+
+        window.addEventListener('resize', function(event){
+            cy.center();
+            cy.resize().fit();
+        });
+        model.on('remove', removeCy)
+    })
+
+    return div
+}
+"""
+
+
+pn.extension("cytoscape", sizing_mode="stretch_width")
+
+elements = [
+    {"data": {"id": "A", "label": "A"}},
+    {"data": {"id": "B", "label": "B"}},
+    {"data": {"id": "A-B", "source": "A", "target": "B"}},
+]
+graph = CytoscapeJS(
+    object=elements,
+    sizing_mode="stretch_width",
+    height=600,
+    styles={"border": "1px solid black"},
+)
+pn.Row(
+    pn.Param(
+        graph,
+        parameters=[
+            "object",
+            "zoom",
+            "pan",
+            "layout",
+            "style",
+            "selected_nodes",
+            "selected_edges",
+        ],
+        sizing_mode="fixed",
+        width=300,
+    ),
+    graph,
+).servable()
+```
+
+:::
+
+:::{tab-item} `ReactComponent`
+
+```{pyodide}
+import param
+import panel as pn
+
+from panel.custom import ReactComponent
+
+
+class CytoscapeReact(ReactComponent):
+
+    object = param.List()
+
+    layout = param.Selector(
+        default="cose",
+        objects=[
+            "breadthfirst",
+            "circle",
+            "concentric",
+            "cose",
+            "grid",
+            "preset",
+            "random",
+        ],
+    )
+    style = param.String("", doc="Use to set the styles of the nodes/edges")
+
+    zoom = param.Number(1, bounds=(1, 100))
+    pan = param.Dict({"x": 0, "y": 0})
+
+    data = param.List(doc="Use to send node's data/attributes to Cytoscape")
+
+    selected_nodes = param.List()
+    selected_edges = param.List()
+
+    _esm = """
+import CytoscapeComponent from 'https://esm.sh/react-cytoscapejs';
+
+export function render({ model }) {
+    function configure(cy){
```

Comment:
I'm in doubt whether this is the correct way to do this or not.

I would have expected the React render function to be rerun whenever a model property changed. I.e. I would not have to configure `model.on`. I would have have expected to be able to set event handlers for `select` and `unselect` events directly on the react `CytoscapeComponent`. But don't know how to do it.

We can leave it as is. But maybe there is a better way that we should be showing of instead.

Reply:
I would have have expected to be able to set event handlers for `select` and `unselect` events directly on the react `CytoscapeComponent`. But don't know how to do it. Maybe its called `tapNode`, `tapNodeData`, `tapEdge`, `tapEdgeData`. See https://github.com/plotly/dash-cytoscape/blob/main/dash_cytoscape/Cytoscape.py.

After searching around I can see others are doing it similar to what we are. For example in https://codesandbox.io/s/react-cytoscape-cl1dxl?file=/src/App.js:5891-5907. Search for `setSelectedNodes` in the example.

Reply:
Fixed.